### ISSUE TITLE
Add support to override build agent NuGet version

### ISSUE THREAD

**Issue by Franklin89** [cake-build/cake-vso#31]
Had the issue on VSTS when i tried to use this extension that a nuget package required a higher nuget client version. Using the powershell integration and running build.ps1 downloaded the latest nuget and the build was successful.

As mentioned on the gitter chat overriding this behavior might be a good solution.

**Comment by gep13** [comment]
@Franklin89 sounds like a good idea to me.  Would you be in a position to do a PR to implement this feature?

**Comment by rj-cooper** [comment]
Sounds like the same thing I came across earlier - there is a PR for it (https://github.com/cake-build/cake-vso/pull/30) but I'm having to wait for legal to OK me to sign the CLA.

**Comment by Franklin89** [comment]
In your PR you can only set the nuget.exe location. I was thinking of something like a selection of "use build agent nuget version" or "use latest nuget version"?

Edit:
Oh i see you could also pass the url and i will be downloaded...

**Comment by rj-cooper** [comment]
It might be worth adding the option to use a nuget.exe already installed on the build agent. Is there a way to find the path to this?

_Edit:_ The vsts-tasks NugetPackager uses

``` csharp
$useBuiltinNuGetExe = !$nuGetPath

if($useBuiltinNuGetExe)
{
    $nuGetPath = Get-ToolPath -Name 'NuGet.exe';
}
```

So we could add an advanced setting for `☑️ Use Build Agent Nuget` which would resolve to something like

```
<AgentInstallFolder>\agent\Worker\Tools\Nuget.exe
```

If that's not checked then fall back to downloading from location provided in NuGet Location or the default of nuget.org

**Comment by Franklin89** [comment]
Sounds like a good idea to me :-)

**Comment by rj-cooper** [comment]
Like this: 
![screenshot_4](https://cloud.githubusercontent.com/assets/28024/19725939/65a96924-9b80-11e6-987a-2fb6a25d2930.png)

**Comment by Franklin89** [comment]
In your PR you can only set the nuget.exe location. I was thinking of something like a selection of "use build agent nuget version" or "use latest nuget version"?

**Comment by devlead** [comment]
Well empty could be nuget in path / build system.

This world give you latest 
https://dist.nuget.org/win-x86-commandline/latest/nuget.exe

And you can also pin a version
https://dist.nuget.org/win-x86-commandline/v3.3.0/nuget.exe

Or use one hosted somewhere else, so think this is a very flexible solution.

See below for reference (nuget.org/nuget.exe isn't recommended anymore)
https://dist.nuget.org/win-x86-commandline/v3.3.0/nuget.exe

**Comment by rj-cooper** [comment]
Do you think it's better to default the `https://nuget.org/nuget.exe` in the UI like I have, or default the UI to empty and do the defaulting behind the scenes (like I have with the `Tool Feed URL`) ?
Either way it's probably better to be consistent with the `Nuget Location` and `Tool Feed URL`

**Comment by devlead** [comment]
What's the current behavior?

**Comment by rj-cooper** [comment]
Current behavior is that within the cake-vso task powershell there is the following

``` powershell
$RootPath = Split-Path -parent $Script;
$ToolPath = Join-Path $RootPath "tools";
$NuGetPath = Join-Path $ToolPath "nuget.exe";
if (!(Test-Path $NuGetPath)) {
  # Download NuGet.exe.
  Write-Verbose "Downloading nuget.exe...";
  (New-Object System.Net.WebClient).DownloadFile("https://nuget.org/nuget.exe", $NuGetPath);
  # Make sure it was properly downloaded.
  if (!(Test-Path $NuGetPath)) {
      Throw "Could not find nuget.exe";
  }
}
```

So by default there is no UI to change the location - it's just hard coded. The original PR https://github.com/cake-build/cake-vso/pull/30 adds the UI by adding an input to the task.json which has a defaultValue e.g.

``` javascript
 "name": "nugetExeLocation",
 "type": "string",
 "label": "Nuget Exe Location",
 "defaultValue": "https://nuget.org/nuget.exe",
 "required": false,
 "helpMarkDown": "Where to fetch the nuget executable from.",
 "groupName": "advanced"
```

![screenshot_5](https://cloud.githubusercontent.com/assets/28024/19739689/dc3b3166-9bb3-11e6-9567-ceea43f93731.png)

Which is then used in the powershell like so:

``` powershell
param(
   [string]$NugetExeLocation,
)
(New-Object System.Net.WebClient).DownloadFile($nugetFolderLocation, $NuGetPath);
```

I think a cleaner alternate would be to set 

``` javascript
 "defaultValue": "",
```

![screenshot_7](https://cloud.githubusercontent.com/assets/28024/19739694/e49a0152-9bb3-11e6-954a-981579a8b6d2.png)

and use

``` powershell
param(
   [string]$NugetExeLocation,
)
if(!$nugetFolderLocation) {
    $nugetFolderLocation = "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
}
(New-Object System.Net.WebClient).DownloadFile($nugetFolderLocation, $NuGetPath);

```

**Comment by devlead** [comment]
Agree empty is nicer, we should document those defaults though.

**Comment by rj-cooper** [comment]
Agreed - I'll add it to the extension-details.md and the hover over help.

**Comment by agc93** [comment]
Fixed by #30

**Comment by agc93** [comment]
Updated extension is now live [on the VS Marketplace](https://marketplace.visualstudio.com/items?itemName=cake-build.cake) including support for selecting NuGet version (thanks @frozenskys !).
